client/cli/src/commands/run_cmd.rs

```diff
@@ -249,6 +249,13 @@ pub struct RunCmd {
 	/// When `--dev` is given and no explicit `--base-path`, this option is implied.
 	#[structopt(long, conflicts_with = "base-path")]
 	pub tmp: bool,
+
+	/// Authenticate authority records not only by the authority key, but also by their peer id.
+	///
+	/// Enable this option if all nodes on the network are already signing their records with the
+	/// peer id.
+	#[structopt(long = "strict-authority-records")]
+	pub strict_authority_records: bool,
```

Comment:
Not sure we want to make this a cli option? Why?

I see this more as particular implementations like Polkadot enable that as they need it.

Reply:
I do not see how Substrate integrations want to upgrade, but without a CLI option, they definitely need 2 updates. One that enables signing, and another one that enables strict checking.

If that is what we need, I can remove the flag from [`Configuration`](https://github.com/paritytech/substrate/blob/9202a478737d916d461bfb0a96a58fbe2a75504a/client/service/src/config.rs#L136), but leave the flag in the [`sc_authority_discovery::WorkerConfig`](https://github.com/paritytech/substrate/blob/9202a478737d916d461bfb0a96a58fbe2a75504a/client/authority-discovery/src/lib.rs#L85), so the [runtime bootstrapping](https://github.com/paritytech/substrate/blob/9202a478737d916d461bfb0a96a58fbe2a75504a/bin/node/cli/src/service.rs#L459) code can enable it.

Reply:
> I can remove the flag from Configuration, but leave the flag in the sc_authority_discovery::WorkerConfig

That was my idea.